Cauldron/Assets/Editor/SODictionaryLoader.cs

```diff
@@ -5,37 +5,52 @@
 namespace Editor
 {
     [InitializeOnLoad]
-    public class SODictionaryLoader
+    [CustomEditor(typeof(SODictionary))]
+    public class SODictionaryLoader : UnityEditor.Editor
     {
+        private static SODictionary targetDictionary;
         static SODictionaryLoader()
         {
             EditorApplication.projectChanged += LoadDictionary;
         }
 
-        static void LoadDictionary()
+        public override void OnInspectorGUI()
         {
-            var dictionary = ScriptableObjectHelper.LoadSingleAsset<SODictionary>();
+            if (GUILayout.Button("Update SODictionary", GUILayout.Height(40)))
+            {
+                targetDictionary = ScriptableObjectHelper.LoadSingleAsset<SODictionary>();
+                
+                targetDictionary.AutoUpdateDictionary = true;
```

Comment:
This is funny, make a private LoadInternal method that skips all validation checks and use it in two places, here and in LoadDictionary(). It's ok to have lots of smaller methods.